### PR TITLE
Harden files drop

### LINKED ISSUE #2083: Add more intergration tests for filesdrop
To avoid bugs like https://github.com/nextcloud/server/pull/2082

We should add an intergration tests for files drop:

1. UserA shared folder `test` via files_drop
2. File 'a.txt' with content 'foo' is uploaded via files_drop
3. File 'a.txt' with content 'bar' is uploaded via files_drop
4. UserA sees `a.txt` and `a (2).txt` in 'test'.
5. `a.txt. contains 'foo'
6. `a (2).txt. contains 'bar'

Assigning to myself so i don't forget.

### PR DESCRIPTION
**Pull request by rullzer**
@LukasReschke as discussed.

Fixes #2083

### CODE REVIEW

File: .htaccess [hunk 1]
```diff
@@ -77,3 +77,7 @@ Options -Indexes
 <IfModule pagespeed_module>
   ModPagespeed Off
 </IfModule>
+#### DO NOT CHANGE ANYTHING ABOVE THIS LINE ####
+
+ErrorDocument 403 //core/templates/403.php
+ErrorDocument 404 //core/templates/404.php
```
Comment (LukasReschke) [inline .htaccess]: 🙈

File: build/integration/features/bootstrap/Sharing.php [hunk 2]
```diff
@@ -185,7 +185,8 @@ public function updatingLastShare($body) {
 		}
 
 		try {
-			$this->response = $client->send($client->createRequest("PUT", $fullUrl, $options));
+			var_dump($fullUrl);
```
Comment (LukasReschke) [inline build/integration/features/bootstrap/Sharing.php]: Debug leftovers :)

**Review by LukasReschke — CHANGES_REQUESTED** [review]

File: build/integration/features/bootstrap/FilesDropContext.php [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+		$client = new Client();
+		$options = [];
+		if (count($this->lastShareData->data->element) > 0){
+			$token = $this->lastShareData->data[0]->token;
+		} else {
+			$token = $this->lastShareData->data[0]->token;
+		}
+
+		$base = substr($this->baseUrl, 0, -4);
+		$fullUrl = $base . '/public.php/webdav/' . $folder . '?XDEBUG_SESSION_START=1';
```
Comment (LukasReschke) [inline build/integration/features/bootstrap/FilesDropContext.php]: XDEBUG_SESSION_START?

File: build/integration/features/bootstrap/Sharing.php [hunk 4]
```diff
@@ -73,7 +73,7 @@ public function asCreatingAShareWith($user, $body) {
 		}
 
 		try {
-			$this->response = $client->send($client->createRequest("POST", $fullUrl, $options));
+			$this->response = $client->send($client->createRequest("POST", $fullUrl . '?XDEBUG_SESSION_START=1', $options));
```
Comment (LukasReschke) [inline build/integration/features/bootstrap/Sharing.php]: X_DEBUG_SESSION_START?

**Review by LukasReschke — DISMISSED** [review]

**Comment by rullzer** [comment]
Fixes https://github.com/nextcloud/server/issues/2083

**Comment by MorrisJobke** [comment]
I fixed the tests - you need to install the instance before running integration tests ;)

**Comment by MorrisJobke** [comment]
I tested the files drop feature and it still works 👍

**Comment by rullzer** [comment]
Aaaah copied the wrong intergration test as example :P

**Review by LukasReschke — APPROVED** [review]
Works nicely! Thanks, @rullzer! 🚀

**Comment by LukasReschke** [comment]
LGTM

**Pull request merged**